libs/vkd3d-shader/dxbc.c

```diff
@@ -2110,13 +2110,13 @@ static int shader_parse_signature(DWORD tag, const char *data, DWORD data_size,
             return VKD3D_ERROR_INVALID_ARGUMENT;
         }
         read_uint32(&ptr, &e[i].semantic_index);
-        read_uint32(&ptr, &e[i].sysval_semantic);
```

Comment:
I wonder if we could do this more easily by just making `read_uint32` take `void*`. Not pretty either, but we're already passing random stuff to it and have to make sure it's actually four bytes in size.

Reply:
ye, `void *` should work as well, might be cleaner.

Reply:
Not having templates is so annoying.

Would be nice to just read a sizeof(T) or static_assert(sizeof(T) == sizeof(uint32_t))